include/cuco/sentinel.hpp

```diff
@@ -21,21 +21,21 @@ namespace sentinel {
 
 template <typename T>
 struct empty_key {
-  __host__ __device__ empty_key(T v) : value{v} {}
+  constexpr empty_key(T v) : value{v} {}
```

Comment:
We should keep the host/device too.

Reply:
Since we are building with `--expt-relaxed-constexpr`, is `__device__` really needed? I have the same question for libcudf as well since we sometimes use `constexpr` with `__device__` but sometimes not.

Reply:
> We should keep the host/device too.

If so, should we rename this file to `sentinel.cuh`?

Reply:
See https://github.com/rapidsai/cudf/issues/7795

In general, I think all functions should be `constexpr`, but I'm not a fan of using `constexpr` to replace `__host__ __device__`.